pkg/pillar/types/assignableadapters.go

```diff
@@ -29,6 +29,73 @@ type AssignableAdapters struct {
 	IoBundleList []IoBundle
 }
 
+type ioBundleError struct {
+	errors      []error
+	timeOfError time.Time
+}
+
+func (iobe *ioBundleError) ErrorTime() time.Time {
+	return iobe.timeOfError
+}
+
+func (iobe *ioBundleError) String() string {
+	if len(iobe.errors) == 0 {
+		return ""
+	}
+	errorStrings := make([]string, 0, len(iobe.errors))
+	for _, err := range iobe.errors {
+		errorStrings = append(errorStrings, err.Error())
+	}
+	return strings.Join(errorStrings, "; ")
+}
+
+func (iobe *ioBundleError) Append(err error) {
+	if iobe.errors == nil {
+		iobe.errors = make([]error, 0, 1)
+	}
+
+	iobe.errors = append(iobe.errors, err)
+
+	iobe.timeOfError = time.Now()
+}
+
+func (iobe *ioBundleError) Empty() bool {
+	if iobe.errors == nil || len(iobe.errors) == 0 {
+		return true
+	}
+
+	return false
+}
+
+func (iobe *ioBundleError) hasError(e error) bool {
+	for _, err := range iobe.errors {
+		if reflect.TypeOf(e) == reflect.TypeOf(err) {
+			return true
+		}
+	}
+
+	return false
+}
+
+func (iobe *ioBundleError) removeByType(e error) {
```

Comment:
```go
func (iobe *ioBundleError) removeByType(e error) {
	filteredErrors := iobe.errors[:0]
	for _, err := range iobe.errors {
		if reflect.TypeOf(e) != reflect.TypeOf(err) {
			filteredErrors = append(filteredErrors, err)
		}
	}
	iobe.errors = filteredErrors
}
```
Isn't it simpler?

Reply:
I don't know... Yes?
Any approach is acceptable while it works. I just thought the second one is more clear. But for sure you can keep your version; I have no problem with it.

Reply:
F.e. if you have a `sync.Mutex` in an error that would fail - but right now I cannot imagine why would anybody need that.

I would prefer to keep it.